server/index.js

```diff
@@ -31,11 +38,32 @@ server.register([require('hapi-auth-cookie'), require('bell')], function (err) {
   // hapi-auth-cookie init
   server.auth.strategy('session', 'cookie', {
     password: config.get('server.session.password'),
-    cookie: config.get('server.session.id'),
-    // TODO temporarily allowing unauthenticated access for visits testing
-    // redirectTo: '/',
+    cookie: config.get('server.session.cookieName'),
+    redirectTo: '/',
     isSecure: config.get('server.session.isSecure'),
-    ttl: config.get('server.session.duration')
+    ttl: config.get('server.session.duration'),
+    clearInvalid: config.get('server.session.clearInvalid'),
+    // this function validates that the user exists + session is valid
+    validateFunc: function(session, cb) {
+      log.verbose('inside hapi-auth-cookie validateFunc.');
+      log.verbose('session is ' + JSON.stringify(session));
+
+      var ttl = config.get('server.session.duration');
+      if (ttl > 0 && new Date() > new Date(session.expiresAt)) {
+        log.verbose('cookie is expired.');
+        return cb(null, false);
+      }
+      log.verbose('cookie is not expired.');
+
+      var fxaId = session.fxaId;
+      db.getUserById(fxaId, function(err, result) {
```

Comment:
Good question. I think this works for now, since the DB is doing basically the same work in either case, but maybe I'll adjust the DB API when we build the `/profile` endpoint. Does that sound good?